Doc/extending/first-extension-module.rst

```diff
@@ -0,0 +1,569 @@
+   >>> status
+   0
+
+.. note::
+
+   The system command ``whoami`` prints out your username.
+   It's useful in tutorials like this one because it has the same name on
+   both Unix and Windows.
+
+
+Warming up your build tool
+==========================
+
+Begin by creating a file named :file:`spammodule.c`. [#why-spammodule]_
+
+Now, while the file is empty, we'll compile it.
+
+Choose a build tool such as Setuptools or Meson, and follow its instructions
+to compile and install the empty :file:`spammodule.c` as a C extension module.
+
+This will ensure that your build tool works, so that you can make
+and test incremental changes as you follow the rest of the text.
+
+.. note:: Workaround for missing ``PyInit``
+
+   If your build tool output complains about missing ``PyInit_spam``,
+   add the following function to your module for now:
+
+   .. code-block:: c
+
+      // A workaround
+      void *PyInit_spam(void) { return NULL; }
+
+   This is a shim for an old-style :ref:`initialization function <extension-export-hook>`,
+   which was required in extension modules for CPython 3.14 and below.
+   Current CPython will not call it, but some build tools may still assume that
+   all extension modules need to define it.
+
+   If you use this workaround, you will get the exception
+   ``SystemError: initialization of spam failed without raising an exception``
+   instead of an :py:exc:`ImportError` in the next step.
+
+.. note::
+
+   Using a third-party build tool is heavily recommended, as in will take
+   care of various details of your platform and Python installation,
+   of naming the resulting extension, and, later, of distributing your work.
+
+   If you don't want to use a tool, you can try to run your compiler directly.
+   The following command should work for many flavors of Linux, and generate
+   a ``spam.so`` file that you need to put in a directory
+   in :py:attr:`sys.path`:
+
+   .. code-block:: sh
+
+      gcc --shared spammodule.c -o spam.so
+
+When your extension is compiled and installed, start Python and try to import
+your extension.
+This should fail with the following exception:
+
+.. code-block:: pycon
+
+   >>> import spam
+   Traceback (most recent call last):
+      ...
+   ImportError: dynamic module does not define module export function (PyModExport_spam or PyInit_spam)
+
+
+Including the Header
+====================
+
+Now, add the first line to your file: include :file:`Python.h` to pull in
+all declarations of the Python C API:
+
+.. literalinclude:: ../includes/capi-extension/spammodule-01.c
+   :start-at: <Python.h>
+   :end-at: <Python.h>
+
+Next, include the header for the :c:func:`system` function:
+
+.. literalinclude:: ../includes/capi-extension/spammodule-01.c
+   :start-at: <stdlib.h>
+   :end-at: <stdlib.h>
+
+Be sure to put this, and any other standard library includes, *after*
+:file:`Python.h`.
+On some systems, Python may define some pre-processor definitions
+that affect the standard headers.
```

Comment:
This is a useful note, but not sure about the tip just below.
It’s giving details that are not needed for this tuto, about something that people shouldn’t do, and already documented in c-api/intro

Reply:
Yes 👍🏽